packages/compass-preferences-model/src/storage.spec.ts

```diff
@@ -41,4 +42,28 @@ describe('storage', function () {
       });
     });
   });
+
+  describe('StoragePreferences', function () {
+    before(function () {
+      tmpDir = mkdtempSync(path.join(os.tmpdir()));
+    });
+
+    after(function () {
+      rmdirSync(tmpDir, { recursive: true });
+    });
+
+    it('will strip unknown prefs', async function () {
+      const prefsStorage = new StoragePreferences(
+        { showedNetworkOptIn: true, foo: true } as unknown as UserPreferences,
+        tmpDir
+      );
+      await prefsStorage.setup();
+
+      // roundabout way to make it load because setup doesn't fill prefsStorage.preferences if it writes the file
```

Comment:
This test is a bit hacky and reliant on knowledge of internals. I think this whole class could benefit from lots more unit tests that would hopefully hilight that sort of thing.